cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/InMemoryStreamMetaStore.java

```diff
@@ -51,10 +55,18 @@ public boolean streamExists(String accountId, String streamName) throws Exceptio
   }
 
   @Override
-  public synchronized List<StreamSpecification> listStreams() throws Exception {
-    ImmutableList.Builder<StreamSpecification> builder = ImmutableList.builder();
-    for (String stream : streams.values()) {
-      builder.add(new StreamSpecification.Builder().setName(stream).create());
+  public synchronized Multimap<NamespaceMeta, StreamSpecification> listStreams() throws Exception {
+    ImmutableMultimap.Builder<NamespaceMeta, StreamSpecification> builder = ImmutableMultimap.builder();
+    for (String accountId : streams.keySet()) {
+      Collection<String> streamNames = streams.get(accountId);
```

Comment:
I think `Collection` returned from a synchronized multimap doesn't provide synchronized iterator. You'll need to sync on the multimap when you iterate the collection returned.